src/Microsoft.Health.Fhir.SqlServer/Features/Search/SqlServerSearchService.cs

```diff
@@ -115,104 +149,107 @@ private async Task<SearchResult> SearchImpl(SearchOptions searchOptions, bool hi
                                                .AcceptVisitor(IncludeRewriter.Instance)
                                            ?? SqlRootExpression.WithDenormalizedExpressions();
 
-            using (var connection = new SqlConnection(_configuration.ConnectionString))
+            using (SqlCommand sqlCommand = connection.CreateCommand())
             {
-                connection.Open();
-
-                using (SqlCommand sqlCommand = connection.CreateCommand())
+                // If we are sending multiple search queries in one transaction
+                if (transaction != null)
                 {
-                    var stringBuilder = new IndentedStringBuilder(new StringBuilder());
+                    // We must assign both the transaction and connection objects to the Command object for a pending local transaction.
+                    sqlCommand.Connection = connection;
```

Comment:
Is this not handled already by the `connection.CreateCommand()` above?

Reply:
Right thanks, makes sense that setting it a second time isn't necessary (I was originally mislead by [this](https://docs.microsoft.com/en-us/dotnet/api/system.data.sqlclient.sqltransaction?view=netcore-2.0)). 

Fixed in 56ad72d